Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM phusion/baseimage
+MAINTAINER Erlang Solutions <mongoose-im@erlang-solutions.com>
+
+RUN useradd -ms /bin/bash amoc
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        git \
+        make \
+        gcc \
+        clang \
+        libexpat1-dev \
+        wget \
+        iproute2 && \
+    wget https://packages.erlang-solutions.com/erlang-solutions_1.0_all.deb && \
+    dpkg -i erlang-solutions_1.0_all.deb && \
+    apt-get update && \
+    apt-get install -y esl-erlang=1:18.3.4
+
+COPY . /amoc_build
+
+RUN cd amoc_build && \
+    make rel && \
+    rm _build/default/rel/amoc/*tar.gz
+
+FROM phusion/baseimage
+MAINTAINER Erlang Solutions <mongoose-im@erlang-solutions.com>
+
+RUN useradd -ms /bin/bash amoc
```

Comment:
duplication?

Reply:
Regarding duplication comments: It is new Docker feature https://docs.docker.com/develop/develop-images/multistage-build/. It allows to create "intermediate" container, which is used just for installing compile time dependencies and compilation. When release is ready inside of temporary container, it is possible to copy just the release into fresh image. It will become final image.